queue-song-lambda/src/main/scala/com/github/saxypandabear/songrequests/spotify/Handler.scala

```diff
@@ -0,0 +1,184 @@
+   */
+  def findActiveComputer(accessToken: String): Option[Device] = {
+    val uriBuilder = new URIBuilder(s"$spotifyApiBaseUrl/devices")
+    val request    = new HttpGet(uriBuilder.build())
+    request.setHeader("Accept", "application/json")
+    request.setHeader("Authorization", s"Bearer $accessToken")
+
+    val httpResponse = httpClient.execute(request)
+    if (httpResponse.getStatusLine.getStatusCode != 200) {
+      logger.error("Something blew up") // TODO: make this more robust
+      None
+    }
+    val responseObj  = objectMapper.readValue(
+        httpResponse.getEntity.getContent,
+        classOf[SpotifyDevicesResponse]
+    )
+    responseObj.devices.find(d => d.isActive && d.deviceType == "Computer")
+  }
+
+  /**
+   * Given a device, queue a Spotify URI to that device
+   * for the given user.
+   * Ref: https://developer.spotify.com/documentation/web-api/reference/#endpoint-add-to-queue
+   * @param accessToken User OAuth access token
+   * @param device      Found active computer device to queue song on
+   * @param spotifyUri  Input Spotify URI
+   */
+  def queueSong(
+      accessToken: String,
+      device: Device,
+      spotifyUri: String
+  ): Unit = {
+    val request = new HttpPost(
+        s"$spotifyApiBaseUrl/queue?uri=$spotifyUri&device_id=${device.id}"
+    )
+    request.setHeader("Accept", "application/json")
+    request.setHeader("Authorization", s"Bearer $accessToken")
+    httpClient.execute(request)
+  }
+
+  /**
+   * Update the refresh token by requesting a new token from Spotify,
+   * and write the updated token into the DynamoDB table.
+   * https://developer.spotify.com/documentation/general/guides/authorization-guide/
+   * @param channelId    The Twitch channel ID (hash key for the DDB table)
+   * @param clientId     Spotify client ID
+   * @param clientSecret Spotify client secret
+   * @param refreshToken Expired refresh token for this user's access
+   */
+  private def refreshSpotifyToken(
+      channelId: String,
+      clientId: String,
+      clientSecret: String,
+      refreshToken: String
+  ): Unit = {
+    val requestBody = Map[String, String](
+        "grant_type"    -> "refresh_token",
+        "refresh_token" -> refreshToken,
+        "client_id"     -> clientId,
+        "client_secret" -> clientSecret
+    )
+    val uriBuilder  = new URIBuilder(oauthUrl)
+    for ((k, v) <- requestBody)
+      uriBuilder.addParameter(k, v)
+    val request  = new HttpPost(uriBuilder.build())
+    request.setHeader("Accept", "application/json")
+    request.setHeader("Content-Type", "application/x-www-form-urlencoded")
+    // TODO: this will become obsolete (hopefully) by migrating to secrets
+    // manager to handle OAuth tokens, so leaving this unfinished for now.
+    val response = httpClient.execute(request)
+  }
+
+  private def initDataStore(
+      projectProperties: ProjectProperties
+  ): ConnectionDataStore =
+    projectProperties.getString("env") match {
```

Comment:
Should move this to the trait. I dislike doing this, but I think it's the cleanest approach. Will do that in a separate PR since it touches other places too